src/cryptonote_core/blockchain.cpp

```diff
@@ -4362,6 +4392,7 @@ bool Blockchain::handle_block_to_main_chain(const block& bl, const crypto::hash&
   get_difficulty_for_next_block(); // just to cache it
   invalidate_block_template_cache();
 
+  send_miner_notifications(id, m_db->get_block_already_generated_coins(m_db->height() - 1));
```

Comment:
Is it not safe to use the `already_generated_coins` already in memory here?

Reply:
I actually didn't see it. I'll test it and if it has the same value I'll change the code.